internal/cli/search/index_opts.go

```diff
@@ -170,14 +165,59 @@ func (opts *IndexOpts) NewSearchIndex() (*atlasv2.SearchIndexCreateRequest, erro
 	return i, nil
 }
 
+func (opts *IndexOpts) readCreateFromFile() (*atlasv2.SearchIndexCreateRequest, error) {
+	index := &atlasv2.SearchIndexCreateRequest{}
+	if err := opts.loadStrict(index); err != nil {
+		return nil, err
+	}
+	if err := file.Load(opts.Fs, opts.Filename, index); err != nil {
+		return nil, fmt.Errorf(failedToLoadIndexMessage, err.Error())
+	}
+
+	if index.Type == nil {
+		index.Type = pointer.Get(DefaultType)
+	}
+
+	return index, nil
+}
+
+func (opts *IndexOpts) loadStrict(v interface{}) error {
```

Comment:
Maybe a better name would be
```suggestion
func (opts *IndexOpts) validateIndexInFile(v interface{}) error {
```

Reply:
agreed! thx

Reply:
it was asked a long time ago by th esearch team for the CLI to not validate since this struct is flexible for them

Reply:
oh ok! removing all the validation then